core/Config/DatabaseConfig.php

```diff
@@ -15,4 +15,9 @@ public static function getSectionName(): string
     {
         return 'database';
     }
+
+    public static function isTiDb(): bool
```

Comment:
Having `isTiDb()` feels odd if we don't also have `isMysql` and `isMariadb`.

If we don't except a lot more checks to come, how about `Schema::supportsLoadDataInfileCharset` or something like that, to match `supportsSortingInSubquery` and `supportsComplexColumnUpdates`?

Reply:
I would consider adding `isMysql` as soon as it is required for something. The `isTidb` will be useful in some other places (like tests) as well. Or do you think it makes more sense to add a new `supports*` method for every special handling we need to TiDb?

Reply:
Yes and no, it depends how many more of those specialties we need to add :sweat_smile:

But that is something we can only say once we know this will be permanent and is complete. And then we may refactor the switches in some way that takes care of it inside the individual vendor classes or whatever we come up with.